internal/sgdisk/sgdisk.go

```diff
@@ -137,6 +145,11 @@ func (op Operation) buildOptions() []string {
 	}
 
 	for _, p := range op.parts {
+		if p.ResizePartition != nil && *p.ResizePartition {
```

Comment:
How does this all work if there are multiple partitions being resized? We're only storing a single `util.PartitionInfo` into the `Operation` object (which is just being overridden every time that `PreservePartitionInfo` is being called in `partitionDisk`). If my config had say 2 partitions being resized wouldn't the current code result in my operation trying to build two different partitions with the same `GUID`, `TypeGUID` & `StartSector` (which are coming from whichever of the two was later in the config)?

Reply:
Yeah, you're right. I think I missed that use case while coding. Let me change the logic to accommodate this requirement.

Reply:
Regardless, I don't think it makes sense to do it this way.  The caller should provide correct information in the `CreatePartition` call, and it's not the responsibility of `sgdisk.go` to override it in some special case.